### ISSUE TITLE
dials.search_beam_position assumes goniometer at 0 position

### ISSUE THREAD

**Issue by Anthchirp** [dials/dials#1164]
follow-up discussion from https://github.com/dials/dials/pull/1160#commitcomment-37372755 onwards

**Comment by rjgildea** [comment]
```
$ dials.show /Users/rjgildea/software/cctbx/modules/dials_regression/indexing_test_data/multi_sweep/SWEEP{1,2}/index/*_datablock_import.json | grep Goniometer -A 3
Goniometer:
    Rotation axis:   {1,-3.94792e-15,7.96523e-15}
    Fixed rotation:  {0.828123,-0.0497396,0.558335,-0.144222,-0.981429,0.126479,0.541675,-0.185264,-0.819918}
    Setting rotation:{1,0,0,0,1,0,0,0,1}
--
Goniometer:
    Rotation axis:   {1,-7.9236e-15,7.97911e-15}
    Fixed rotation:  {0.828123,-0.458662,-0.322243,-0.144222,0.381181,-0.913182,0.541675,0.802702,0.249516}
    Setting rotation:{1,0,0,0,1,0,0,0,1}
```

Behaviour before initial refactoring to experiments:
```
$ git checkout 17f1fd169
```
```
$ dials.search_beam_position /Users/rjgildea/software/cctbx/modules/dials_regression/indexing_test_data/multi_sweep/SWEEP{1,2}/index/*_{datablock_import.json,strong.pickle}
...
Running DPS using 281 reflections
Running DPS using 316 reflections
Found 7 solutions with max unit cell 7.55 Angstroms.
Found 7 solutions with max unit cell 7.55 Angstroms.
Old beam centre: 39.12, 97.87 mm (227.4, 569.2 px)
New beam centre: 41.43, 98.28 mm (240.9, 571.5 px)
Shift: -2.31, -0.40 mm (-13.4, -2.3 px)

Saving optimised experiments to optimised.expt
```

Behaviour after refactoring:
```
$ git checkout 7879db4c1
```
```
$ dials.search_beam_position /Users/rjgildea/software/cctbx/modules/dials_regression/indexing_test_data/multi_sweep/SWEEP{1,2}/index/*_{datablock_import.json,strong.pickle}
...
Running DPS using 281 reflections
Running DPS using 316 reflections
Found 7 solutions with max unit cell 7.55 Angstroms.
Found 7 solutions with max unit cell 7.55 Angstroms.
Old beam centre: 39.12, 97.87 mm (227.4, 569.2 px)
New beam centre: 43.54, 100.21 mm (253.1, 582.8 px)
Shift: -4.42, -2.34 mm (-25.7, -13.6 px)

Saving optimised experiments to optimised.expt
```

**Comment by rjgildea** [comment]
```
$ dials.import /dls/i03/data/2019/cm23003-4/20190911/SmarGon/rotation_calibration*/*_master.h5
$ dials.find_spots imported.expt nproc=24
$ dials.split_experiments imported.expt strong.refl
$ dials.combine_experiments split_{1,2}.*
$ dials.show combined.expt | grep Goniometer -A 8
Goniometer:
    Rotation axis:   {1,0,0}
    Fixed rotation:  {0.999989,-0.00469193,-0.000127593,0.00322752,0.707109,-0.707098,0.00340787,0.707089,0.707116}
    Setting rotation:{1,0,0,0,1,0,0,0,1}
    Axis #0 (phi):  {1,-0.0025,0.0056}
    Axis #1 (chi):  {-0.006,-0.0264,-0.9996}
    Axis #2 (omega):  {1,0,0}
    Angles: 45,0,0
    scan axis: #2 (omega)
--
Goniometer:
    Rotation axis:   {1,0,0}
    Fixed rotation:  {0.709333,0.484575,-0.511892,-0.70447,0.511928,-0.491582,0.0238432,0.709307,0.704496}
    Setting rotation:{1,0,0,0,1,0,0,0,1}
    Axis #0 (phi):  {1,-0.0025,0.0056}
    Axis #1 (chi):  {-0.006,-0.0264,-0.9996}
    Axis #2 (omega):  {1,0,0}
    Angles: 45,45,0
    scan axis: #2 (omega)
```

With current release:
```
$ module load dials
$ time dials.search_beam_position combined.*
DIALS (2018) Acta Cryst. D74, 85-97. https://doi.org/10.1107/S2059798317017235
The following parameters have been modified:

input {
  experiments = combined.expt
  reflections = combined.refl
}

Selecting subset of 10000 reflections for analysis
Selecting subset of 10000 reflections for analysis
Running DPS using 10000 reflections
Found 22 solutions with max unit cell 199.33 Angstroms.
Running DPS using 10000 reflections
Found 19 solutions with max unit cell 199.33 Angstroms.
Old beam centre: 156.31, 169.37 mm (2084.1, 2258.3 px)
New beam centre: 156.41, 169.86 mm (2085.5, 2264.8 px)
Shift: -0.11, -0.48 mm (-1.4, -6.4 px)

Saving optimized experiments to optimised.expt

real	1m46.221s
user	1m26.820s
sys	0m3.789s
```

With latest nightly build:
```
$ module load dials/nightly
$ time dials.search_beam_position combined.*
DIALS (2018) Acta Cryst. D74, 85-97. https://doi.org/10.1107/S2059798317017235
The following parameters have been modified:

input {
  experiments = combined.expt
  reflections = combined.refl
}

Selecting subset of 10000 reflections for analysis
Selecting subset of 10000 reflections for analysis
Running DPS using 10000 reflections
Running DPS using 10000 reflections
Found 22 solutions with max unit cell 199.33 Angstroms.
Found 19 solutions with max unit cell 199.33 Angstroms.
Sorry: No valid scores

real	3m36.907s
user	3m5.435s
sys	0m4.746s
```
